app/controllers/concerns/api_user_authentication.rb

```diff
@@ -0,0 +1,30 @@
+module APIUserAuthentication
```

Comment:
It's slightly confusing that this is one of two authentication methods for APIs. Could this maybe be `ServiceAPIUserAuthentication` or something? Is a "service API" a thing?